### PR TITLE
ML benchmark playbooks

### PR DESCRIPTION
**Pull request by kpouget**
Follow up of #215 (Pam's work) and https://github.com/openshift-psap/ci-artifacts/pull/222 (Coco dataset download example)

### CODE REVIEW

**Comment by kpouget** [comment]
@omertuc the mirror seems to reject the certificate:
```
+ md5sum /etc/mirror-secret/mirror.pem
df81f0d8994c7dac9632ddd2a07727fe  /etc/mirror-secret/mirror.pem
+ [[ -e train2017.zip ]]
+ [[ ! -e train2017.zip ]]
+ curl -O -E /etc/mirror-secret/mirror.pem https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco/train2017.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   153  100   153    0     0    254      0 --:--:-- --:--:-- --:--:--   253

real	0m0.610s
user	0m0.020s
sys	0m0.005s
+ md5sum train2017.zip
7d6137132ff4e08f6848aee8c3ea20e8  train2017.zip
```

locally I get the same:
```
$ curl -E openshift/.creds/entitled-mirror-client-creds.pem https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco/train2017.zip -s | md5sum
7d6137132ff4e08f6848aee8c3ea20e8  -

$ curl -E openshift/.creds/entitled-mirror-client-creds.pem https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco/train2017.zip -sSf
curl: (22) The requested URL returned error: 403
```
(I'll update the PR to use `-sSf` for curl ^^)

and I double checked, the PEM is the one from vault:
```
df81f0d8994c7dac9632ddd2a07727fe  /var/run/psap-entitlement-secret/entitled-mirror-client-creds.pem
```
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-psap-ci-artifacts-release-4.9-gpu-operator-e2e-master/1424505781717307392/artifacts/gpu-operator-e2e-master/nightly/build-log.txt

**Comment by omertuc** [comment]
> (I'll update the PR to use `-sSf` for curl ^^)

Please use the long form for those flags when you do

> @omertuc the mirror seems to reject the certificate:

Weird, looking into it

**Comment by omertuc** [comment]
> > (I'll update the PR to use `-sSf` for curl ^^)
> 
> Please use the long form for those flags when you do
> 
> > @omertuc the mirror seems to reject the certificate:
> 
> Weird, looking into it

Restarting the deployment fixed the issue. Looks like the pod lost access to /mirror_files/ volume mount . Might be some weird Amazon EBS quirk

**Comment by kpouget** [comment]
thanks @omertuc , we'll see if that happens periodically or if it was only a one-shot weirdness

/test gpu-operator-commit

**Comment by kpouget** [comment]
```
2021-08-10 12:29:11,800 p=60 u=psap-ci-runner n=ansible | <stderr> Error from server: Get "https://10.0.138.191:10250/containerLogs/default/coco-dataset-downloader/coco-dataset-downloader": x509: certificate signed by unknown authority
```
I wonder what that can be, all the log fetching fail because of that, see [gather-extra logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1425069085598683136/artifacts/gpu-operator-commit/gather-extra/build-log.txt)

/test gpu-operator-commit

**Comment by kpouget** [comment]
```
+ cd /storage
+ DATASET_FILENAME=train2017.zip
+ [[ -z https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco ]]
+ [[ ! -e /etc/mirror-secret/mirror.pem ]]
+ md5sum /etc/mirror-secret/mirror.pem
df81f0d8994c7dac9632ddd2a07727fe  /etc/mirror-secret/mirror.pem
+ [[ -e train2017.zip ]]
+ [[ ! -e train2017.zip ]]
+ curl -O -E /etc/mirror-secret/mirror.pem https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco/train2017.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0 18.0G    0  192k    0     0   162k      0 32:17:32  0:00:01 32:17:31  162k
  0 18.0G    0 7855k    0     0  3709k      0  1:24:51  0:00:02  1:24:49 3707k
  0 18.0G    0 16.7M    0     0  5536k      0  0:56:50  0:00:03  0:56:47 5535k
  0 18.0G    0 27.2M    0     0  6753k      0  0:46:36  0:00:04  0:46:32 6751k
  0 18.0G    0 37.7M    0     0  7595k      0  0:41:26  0:00:05  0:41:21 7774k
  0 18.0G    0 41.1M    0     0  6961k      0  0:45:12  0:00:06  0:45:06 8611k
curl: (18) transfer closed with 19293722744 bytes remaining to read

real	0m6.061s
user	0m0.060s
sys	0m0.103s
```
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1425101673579155456/artifacts/gpu-operator-commit/presubmit-commit/artifacts/000__benchmarking__download_coco_dataset/pod_downloader.log

@omertuc could you check the logs of the mirror to see if there is anything explaining this issue?

**Comment by omertuc** [comment]
> ```
> + cd /storage
> + DATASET_FILENAME=train2017.zip
> + [[ -z https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco ]]
> + [[ ! -e /etc/mirror-secret/mirror.pem ]]
> + md5sum /etc/mirror-secret/mirror.pem
> df81f0d8994c7dac9632ddd2a07727fe  /etc/mirror-secret/mirror.pem
> + [[ -e train2017.zip ]]
> + [[ ! -e train2017.zip ]]
> + curl -O -E /etc/mirror-secret/mirror.pem https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco/train2017.zip
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
> 
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
>   0 18.0G    0  192k    0     0   162k      0 32:17:32  0:00:01 32:17:31  162k
>   0 18.0G    0 7855k    0     0  3709k      0  1:24:51  0:00:02  1:24:49 3707k
>   0 18.0G    0 16.7M    0     0  5536k      0  0:56:50  0:00:03  0:56:47 5535k
>   0 18.0G    0 27.2M    0     0  6753k      0  0:46:36  0:00:04  0:46:32 6751k
>   0 18.0G    0 37.7M    0     0  7595k      0  0:41:26  0:00:05  0:41:21 7774k
>   0 18.0G    0 41.1M    0     0  6961k      0  0:45:12  0:00:06  0:45:06 8611k
> curl: (18) transfer closed with 19293722744 bytes remaining to read
> 
> real	0m6.061s
> user	0m0.060s
> sys	0m0.103s
> ```
> 
> [gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1425101673579155456/artifacts/gpu-operator-commit/presubmit-commit/artifacts/000__benchmarking__download_coco_dataset/pod_downloader.log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1425101673579155456/artifacts/gpu-operator-commit/presubmit-commit/artifacts/000__benchmarking__download_coco_dataset/pod_downloader.log)
> 
> @omertuc could you check the logs of the mirror to see if there is anything explaining this issue?

image

**Comment by omertuc** [comment]
I'm guessing you'll need to use heavier machinery to download such a large file, something other than curl, or `curl` but with re-tries and picking up a connection where it left off incase it dies abruptly

**Comment by kpouget** [comment]
thanks for the logs,

yes, if client and server don't know why the connection was closed, then
we'll need to improve the downloader code ...

weird that it worked at the first try last time, must be luck :#

**Comment by kpouget** [comment]
[test failed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1425796205010489344/artifacts/gpu-operator-commit/presubmit-commit/artifacts/000__benchmarking__download_coco_dataset/pod_downloader.log) because only 890MB of data was downloaded before reaching the retry count. Increasing it to 20min



/test gpu-operator-commit

**Comment by kpouget** [comment]
```
            "message": "0/7 nodes are available: 1 node(s) had volume node affinity conflict, 3 Insufficient nvidia.com/gpu, 3 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate.",
```

**Comment by kpouget** [comment]
\o/
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1434121308366442496/artifacts/gpu-operator-commit/presubmit-commit/artifacts/013__benchmarking__run_nvidiadl_ssd/nvidiadl_ssd_pod.log
@omertuc @ArangoGutierrez @dagrayvid PR is ready for review, let me know if you have any comment

\hold <-- there's still a `WIP` commit for triggering the benchmark

File: roles/benchmarking_deploy_coco_dataset/tasks/main.yml
Comment (omertuc) [inline roles/benchmarking_deploy_coco_dataset/tasks/main.yml:9]: Consider a more idempotent approach, or protect this command with an `--allow-overwrite` toolbox flag

Comment (omertuc) [reply roles/benchmarking_deploy_coco_dataset/tasks/main.yml:9]: Oh sorry, you're just deleting the secret, not the mirror. Ignore me

Comment (kpouget) [reply roles/benchmarking_deploy_coco_dataset/tasks/main.yml:9]: hum, you though it was the PVC that was being deleted I guess ;-)
no, no command doing that for now ;-)

the test is actually validating that the command is idempotent, and doesn't  download twice the dataset (and no more md5sum during the 2nd try, only a simple flag-file check ;-))
```
for i in 1 2; do
    ./run_toolbox.py benchmarking download_coco_dataset \
                     "$gpu_node_hostname" \
                     --mirror_base_url=https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco \
                     --client-cert=${PSAP_SECRET_PATH}/entitled-mirror-client-creds.pem
done
```

**Comment by kpouget** [comment]
/hold

I'll reorder the commit and get the test commited-and-reverted with https://github.com/openshift-psap/ci-artifacts/pull/251

**Comment by kpouget** [comment]
Hello @omertuc @dagrayvid can get merge this PR?

The tests show that:
1. the coco dataset is properly [downloaded](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/011__benchmarking__download_coco_dataset/) from the CI mirror
2. calling this command twice has [no side effect](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/012__benchmarking__download_coco_dataset/pod_downloader.log), the dataset is not re-downloaded, and checking it takes only [1min](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/012__benchmarking__download_coco_dataset/_ansible.log) as I use a file-flag `$FILENAME.extracted` to ensure that the file was correctly downloaded (`md5sum` checked) and extracted. This file-flag avoid computing the md5sum of big files that takes several minutes to complete
3. the SSD benchmark is executed (8min), its Pod logs are [stored](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/013__benchmarking__run_nvidiadl_ssd/nvidiadl_ssd_pod.log),  as well as the execution [status](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/013__benchmarking__run_nvidiadl_ssd/status), in order to simplify post-processing in the future.

Overall, what is missing here is that the benchmark pod image is [hard-coded](https://github.com/openshift-psap/ci-artifacts/pull/235/files#diff-5c81f9296951e90942fc06aed989a0c7f435b53f5d4025680a5c4105ebd24637R12) (`quay.io/openshift-psap/nvidiadl-ssd-training-benchmark`) and was manually generated, but this will be solved later on.

Note that the 8-minute execution time includes the image downloading (4.5GB).

**Comment by kpouget** [comment]
> Hello @omertuc @dagrayvid can get merge this PR?
> 
> The tests show that:
> 
>     1. the coco dataset is properly [downloaded](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/011__benchmarking__download_coco_dataset/) from the CI mirror
> 
>     2. calling this command twice has [no side effect](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/012__benchmarking__download_coco_dataset/pod_downloader.log), the dataset is not re-downloaded, and checking it takes only [1min](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/012__benchmarking__download_coco_dataset/_ansible.log) as I use a file-flag `$FILENAME.extracted` to ensure that the file was correctly downloaded (`md5sum` checked) and extracted. This file-flag avoid computing the md5sum of big files that takes several minutes to complete
> 
>     3. the SSD benchmark is executed (8min), its Pod logs are [stored](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/013__benchmarking__run_nvidiadl_ssd/nvidiadl_ssd_pod.log),  as well as the execution [status](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/013__benchmarking__run_nvidiadl_ssd/status), in order to simplify post-processing in the future.
> 
> 
> Overall, what is missing here is that the benchmark pod image is [hard-coded](https://github.com/openshift-psap/ci-artifacts/pull/235/files#diff-5c81f9296951e90942fc06aed989a0c7f435b53f5d4025680a5c4105ebd24637R12) (`quay.io/openshift-psap/nvidiadl-ssd-training-benchmark`) and was manually generated, but this will be solved later on.
> 
> Note that the 8-minute execution time includes the image downloading (4.5GB).

Hello @jmencak, would you have a moment to have a look at this PR?
it's not complete yet as it's not included in any nightly run, but that will be follow-up work for someone else!

File: roles/benchmarking_deploy_coco_dataset/files/entrypoint.sh
Comment (jmencak) [inline roles/benchmarking_deploy_coco_dataset/files/entrypoint.sh:30]: Do we also want to `exit 1` on failure of installing `unzip`?

File: roles/benchmarking_deploy_coco_dataset/files/entrypoint.sh
Comment (jmencak) [inline roles/benchmarking_deploy_coco_dataset/files/entrypoint.sh:42]: A future PR should probably cleanup all these `echo "INFO/FATAL: " into simple functions such as info/fatal()` and also probably log to `STDERR` vs. `STDOUT`.

File: roles/benchmarking_deploy_coco_dataset/meta/main.yml
Comment (jmencak) [inline roles/benchmarking_deploy_coco_dataset/meta/main.yml:3]: Github helps you to give you a little nudge here.  Most files with no EOL are a no-no in my view.

File: roles/benchmarking_run_nvidiadl_ssd/tasks/main.yml
Comment (jmencak) [inline roles/benchmarking_run_nvidiadl_ssd/tasks/main.yml:14]: `s/it it/if it/`

File: roles/benchmarking_run_nvidiadl_ssd/tasks/main.yml
Comment (jmencak) [inline roles/benchmarking_run_nvidiadl_ssd/tasks/main.yml:60]: `s/  / /` (double space vs. single)

File: roles/benchmarking_run_nvidiadl_ssd/templates/nvidiadl-ssd-pod.yml.j2
Comment (jmencak) [inline roles/benchmarking_run_nvidiadl_ssd/templates/nvidiadl-ssd-pod.yml.j2:9]: `s/e}/e }/`

File: roles/benchmarking_deploy_coco_dataset/templates/pod.yml.j2
Comment (jmencak) [inline roles/benchmarking_deploy_coco_dataset/templates/pod.yml.j2:9]: `s/e}/e }/`

**Review by jmencak — COMMENTED** [review]
It is hard for me to understand the PR fully without testing and playing with it a bit.  Nevertheless, the tests pass the code seems sane.  Only nits.  Once addressed, I'm happy to approve.

File: roles/benchmarking_deploy_coco_dataset/templates/pod.yml.j2
Comment (kpouget) [reply roles/benchmarking_deploy_coco_dataset/templates/pod.yml.j2]: +1
this one should have been caught by `ansible-lint` .... :/

File: roles/benchmarking_run_nvidiadl_ssd/tasks/main.yml
Comment (kpouget) [reply roles/benchmarking_run_nvidiadl_ssd/tasks/main.yml]: +1

File: roles/benchmarking_run_nvidiadl_ssd/tasks/main.yml
Comment (kpouget) [reply roles/benchmarking_run_nvidiadl_ssd/tasks/main.yml]: +1
should have been caught by `ansible-lint` again :/ :/

File: roles/benchmarking_run_nvidiadl_ssd/templates/nvidiadl-ssd-pod.yml.j2
Comment (kpouget) [reply roles/benchmarking_run_nvidiadl_ssd/templates/nvidiadl-ssd-pod.yml.j2]: damn copy and paste :#

File: roles/benchmarking_deploy_coco_dataset/defaults/main/config.yml
Comment (kpouget) [reply roles/benchmarking_deploy_coco_dataset/defaults/main/config.yml]: good catch, I don't think it changes anything. Will fix it for consistency

**Review by kpouget — COMMENTED** [review]
thanks @jmencak for the review, all the comments should have been fixed with the last push

/hold <-- added back the TEST route path to validate the last changes + rebase

**Comment by jmencak** [comment]
/lgtm
/hold
Please cancel the hold once you're happy with the testing, Kevin.

**Comment by kpouget** [comment]
> /lgtm
> /hold
> Please cancel the hold once you're happy with the testing, Kevin.

thanks a lot for the review @jmencak :)

--- 

test failed after rebase, the GPU Operator now needs `--namespace openshift-operators` to be deployed.

testing on 4.6 as 4.7 and 4.8 seems to be affected by the missing kernel package, so the driver cannot be built.

/test 4.6-gpu-operator-e2e

File: build/root/usr/local/bin/ci_entrypoint_custom.sh
Comment (jmencak) [inline build/root/usr/local/bin/ci_entrypoint_custom.sh:18]: The output will be messy on SNO clusters for example, where you have dual (master,worker) roles separated with `','`.  If that's not a problem to you, happy to `/lgtm` again.

**Comment by kpouget** [comment]
>  ci/prow/4.6-gpu-operator-e2e — Job failed.                    
```
* could not run steps: step gpu-operator-e2e failed: failed to wait for the created cluster claim to become ready: timed out waiting for the condition 
```
:( 

trying again, but maybe we're requestiong 4.6 clusters from a non-existing pool (or pool is just empty, we'll see)

/test 4.6-gpu-operator-e2e

File: build/root/usr/local/bin/ci_entrypoint_custom.sh
Comment (kpouget) [reply build/root/usr/local/bin/ci_entrypoint_custom.sh:18]: good point :)
this file `ci_entrypoint_custom.sh` will be reverted once the PR is merged, so not an issue here

**Comment by kpouget** [comment]
I'll wait for this PR to be merged: https://github.com/openshift-psap/ci-artifacts/pull/255
it will offer a way to easily add and trigger custom testing, similarly to what this PR is doing on its top commit

**Comment by kpouget** [comment]
the ci-mirror server wasn't feeling very well (high memory pressure) ...

should be better now, hopefully

/test custom

**Comment by kpouget** [comment]
> /lgtm
> /hold
> Please cancel the hold once you're happy with the testing, Kevin.

@jmencak PR is now good for merging, 
the only think that changed since your previous `/lgtm` is the location of the test script `[testing/custom/run_cocodataset_nvidiadl_ssd.sh](https://github.com/openshift-psap/ci-artifacts/blob/f72b7dd5749827fa01b915e66633f2eabb9a7de2/testing/custom/run_cocodataset_nvidiadl_ssd.sh)` which now belongs to `testing/custom` and will stay there for future reference & testing

File: roles/benchmarking_deploy_coco_dataset/tasks/main.yml
Comment (jmencak) [inline roles/benchmarking_deploy_coco_dataset/tasks/main.yml:13]: `--ignore-not-found` is true by default, no need for `--ignore-not-found=true`
Similarly below.

File: roles/benchmarking_deploy_coco_dataset/tasks/main.yml
Comment (jmencak) [inline roles/benchmarking_deploy_coco_dataset/tasks/main.yml:79]: Extra new line.

**Review by jmencak — APPROVED** [review]
/lgtm
Thank you Kevin, looks good, only nits that can be fixed later on.

File: roles/benchmarking_deploy_coco_dataset/tasks/main.yml
Comment (kpouget) [reply roles/benchmarking_deploy_coco_dataset/tasks/main.yml:13]: oh you mean `--ignore-not-found=true` is like `test == true`?
true indeed :) but there are 46 places where it should be updated :#
will be for another PR!

**Comment by kpouget** [comment]
> /lgtm
> Thank you Kevin, looks good, only nits that can be fixed later on.

thanks @jmencak :)

**Pull request merged**
